pii_recognizer/pii_recognizer.py

```diff
@@ -481,72 +475,180 @@ def _get_analyzer_engine(model="whole"):
         spacy_recognizer = CustomSpacyRecognizer()
         # add the custom build spacy recognizer
         registry.add_recognizer(spacy_recognizer)
-    if model == "flair":
+    elif model == "flair":
         # pre-trained flair recognizer
         flair_recognizer = FlairRecognizer()
         # add the custom build flair recognizer
         registry.add_recognizer(flair_recognizer)
-    if model == "pattern":
+    elif model == "pattern":
         # add the pattern recognizer
         pattern_recognizer_factory = PatternRecognizerFactory()
-        for recognizer in pattern_recognizer_factory.create_pattern_recognizer():
+        for recognizer in pattern_recognizer_factory._create_pattern_recognizer():
             registry.add_recognizer(recognizer)
-    if model == "whole":
+    elif model == "whole":
         spacy_recognizer = CustomSpacyRecognizer()
         flair_recognizer = FlairRecognizer()
         registry.add_recognizer(spacy_recognizer)
         registry.add_recognizer(flair_recognizer)
         # add the pattern recognizer
         pattern_recognizer_factory = PatternRecognizerFactory()
-        for recognizer in pattern_recognizer_factory.create_pattern_recognizer():
+        for recognizer in pattern_recognizer_factory._create_pattern_recognizer():
             registry.add_recognizer(recognizer)
+    elif not model and entities:
+        if set(entities).intersection(
+            set(["LOCATION", "PERSON", "NRP", "ORGANIZATION", "DATE_TIME"])
```

Comment:
I think it will be better first to have a constants class for recognizers (aka models) and entities like:

```python
class Models:
    WHOLE = "whole"
    PATTERN = "pattern"
    FLAIR = "flair"


class Entities:
    LOCATION = "LOCATION"
    PERSON = "PERSON"
    ...
```

In addition, each recognizer should add a `set` class attribute that marks what entities it knows to recognize so you can take to user input and do:

```python
# Add a new class attribute to a Recognizer:
class CustomSpacyRecognizer:
    RECOGNIZABLE_ENTITIES = {
        Entities.LOCATION,
        Entities.PERSON,
        ...
    }

# So later here in code you can do (the `&` operator is instersection of sets):
if set(entities) & CustomSpacyRecognizer.RECOGNIZABLE_ENTITIES:
    ...
```